pkg/cstor/pool/v1alpha2/create.go

```diff
@@ -0,0 +1,92 @@
+package v1alpha2
+
+import (
+	"github.com/golang/glog"
+	api "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha2"
+	zfs "github.com/openebs/maya/pkg/zfs/cmd/v1alpha1"
+)
+
+// Create will create the pool for given csp object
+func Create(csp *api.CStorNPool) error {
+	var err error
+	raidGroups := csp.Spec.RaidGroups
+
+	glog.Infof("Creating a pool for %s %s", csp.Name, PoolName(csp))
+
+	// First create a pool
+	// TODO, IsWriteCache, IsSpare, IsReadCache should be disable for actual pool?
+
+	// Lets say we need to execute following command
+	// -- zpool create newpool mirror v0 v1 mirror v2 v3 log mirror v4 v5
+	// Above command we will execute using following steps:
+	// 1. zpool create newpool mirror v0 v1
+	// 2. zpool add newpool log mirror v4 v5
+	// 3. zpool add newpool mirror v2 v3
+	for i, r := range raidGroups {
+		if !r.IsReadCache && !r.IsSpare && !r.IsWriteCache {
+			// we found the main raidgroup. let's create the pool
+			err := createPool(csp, &r)
+			if err != nil {
+				glog.Errorf("Failed to create pool {%s} : %s", PoolName(csp), err.Error())
+				return err
+			}
+			// Remove this raidGroup
+			raidGroups = append(raidGroups[:i], raidGroups[i+1:]...)
+			break
+		}
+	}
+
+	// We created the pool
+	// Lets update it with extra config, if provided
+	for _, r := range raidGroups {
+		if e := addRaidGroup(csp, &r); e != nil {
+			err = ErrorWrapf(err, "Failed to add raidGroup{%s}.. %s", r.Name, e.Error())
+		}
+	}
+
+	// TODO, should we delete the pool?
```

Comment:
reconcilation should ideally take care of this..

Reply:
even updating devLinks in CSP

Reply:
ok